sklearn/linear_model/stochastic_gradient.py

```diff
@@ -66,6 +67,29 @@ def __init__(self, loss, penalty='l2', alpha=0.0001, C=1.0,
         self.warm_start = warm_start
         self.average = average
 
+        if n_iter is not None:
+            warnings.warn("n_iter parameter is deprecated in 0.19 and will be"
+                          " removed in 0.21. Use max_iter and tol instead.",
+                          DeprecationWarning)
+            # Same behavior as before 0.19
+            self.max_iter = n_iter
+            self.tol = -np.inf
```

Comment:
Also please add a check in `_validate_params` that raises `ValueError` if `self.tol < 0` telling that `tol` should be positive (or zero to disable the convergence criterion).